testing/integration/agent_long_running_leak_test.go

```diff
@@ -0,0 +1,296 @@
+		MonitoringEnabled: []kibana.MonitoringEnabledOption{
+			kibana.MonitoringEnabledLogs,
+			kibana.MonitoringEnabledMetrics,
+		},
+	}
+
+	policyResp, err := tools.InstallAgentWithPolicy(ctx, runner.T(), installOpts, runner.agentFixture, runner.info.KibanaClient, basePolicy)
+	require.NoError(runner.T(), err)
+
+	// install system package
+	runner.InstallPackage(ctx, "system", "1.53.1", "agent_long_test_base_system_integ.json", uuid.New().String(), policyResp.ID)
+
+	// install cef
+	runner.InstallPackage(ctx, "apache", "1.17.0", "agent_long_test_apache.json", uuid.New().String(), policyResp.ID)
+
+}
+
+func (runner *ExtendedRunner) InstallPackage(ctx context.Context, name string, version string, cfgFile string, policyUUID string, policyID string) {
+	installPackage := kibana.PackagePolicyRequest{}
+
+	jsonRaw, err := os.ReadFile(cfgFile)
+	require.NoError(runner.T(), err)
+
+	err = json.Unmarshal(jsonRaw, &installPackage)
+	require.NoError(runner.T(), err)
+
+	installPackage.Package.Version = version
+	installPackage.ID = policyUUID
+	installPackage.PolicyID = policyID
+	installPackage.Namespace = "default"
+	installPackage.Name = fmt.Sprintf("%s-long-test-%s", name, policyUUID)
+	installPackage.Vars = map[string]interface{}{}
+
+	runner.T().Logf("Installing %s package....", name)
+	_, err = runner.info.KibanaClient.InstallFleetPackage(ctx, installPackage)
+	require.NoError(runner.T(), err, "error creating fleet package")
+}
+
+func (runner *ExtendedRunner) TestHandleLeak() {
+	ctx, cancel := context.WithTimeout(context.Background(), time.Hour)
+	defer cancel()
+
+	testRuntime := os.Getenv("LONG_TEST_RUNTIME")
+	if testRuntime == "" {
+		testRuntime = "15m"
+	}
+
+	regex := regexp.MustCompile(`[\d]+`)
+	status, err := runner.agentFixture.ExecStatus(ctx)
+	require.NoError(runner.T(), err)
+
+	// because we need to separately fetch the PIDs, wait until everything is healthy before we look for running beats
+	require.Eventually(runner.T(), func() bool {
+		allHealthy := true
+		status, err := runner.agentFixture.ExecStatus(ctx)
+
+		cefMatch := "logfile-apache"
+		foundCef := false
+		systemMatch := "metrics-default"
+		foundSystem := false
+
+		require.NoError(runner.T(), err)
+		for _, comp := range status.Components {
+			// make sure the components include the expected integrations
+			for _, v := range comp.Units {
+				runner.T().Logf("unit ID: %s", v.UnitID)
+				// the full unit ID will be something like "log-default-logfile-cef-3f0764f0-4ade-4f46-9ead-f2f0f7865676"
+				if !foundCef && strings.Contains(v.UnitID, cefMatch) {
+					foundCef = true
+				}
+				if !foundSystem && strings.Contains(v.UnitID, systemMatch) {
+					foundSystem = true
+				}
+			}
+			runner.T().Logf("component state: %s", comp.Message)
+			if comp.State != int(cproto.State_HEALTHY) {
+				allHealthy = false
+			}
+		}
+		return allHealthy && foundCef && foundSystem
+	}, time.Minute*3, time.Second*20, "install never became healthy")
```

Comment:
These duration could be set in the ExtendedRunner so we can make it easier to change